### PR TITLE
Disable txpool if SequencerUrl is set

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Disable txpool if SequencerUrl is set or it does a bunch of busy work on NewHead/HeadChange, AddPeer

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs [hunk 1]
```diff
@@ -54,6 +54,11 @@ protected override void RegisterEthRpcModule(IRpcModuleProvider rpcModuleProvide
             _logger.Warn($"SequencerUrl is not set. Nethermind will behave as a Sequencer");
         }
 
+        if (_config.SequencerUrl is not null)
+        {
+            _api.TxPool.Disable();
+        }
+
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs:61]: Maybe better not to create it in the first place or create some NullTxPool?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs:61]: There is already `InitializeBlockchainOptimism.cs` so it is easy to override there

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs:61]: We do have a `NullTxPool`

**Comment by benaadams** [comment]
Looks good

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs [hunk 2]
```diff
@@ -51,13 +51,12 @@ protected override void RegisterEthRpcModule(IRpcModuleProvider rpcModuleProvide
...
         if (_config.SequencerUrl is null && _logger.IsWarn)
         {
-            _logger.Warn($"SequencerUrl is not set. Nethermind will behave as a Sequencer");
+            _logger.Warn("SequencerUrl is not set. Nethermind will behave as a Sequencer");
         }
 
-        BasicJsonRpcClient? sequencerJsonRpcClient = _config.SequencerUrl is null
-            ? null
-            : new(new Uri(_config.SequencerUrl), _api.EthereumJsonSerializer, _api.LogManager);
-        ModuleFactoryBase<IEthRpcModule> ethModuleFactory = CreateEthModuleFactory();
```
Comment (jmederosalvarado) [inline src/Nethermind/Nethermind.Optimism/Rpc/RegisterOptimismRpcModules.cs:60]: why delete the ModuleFactoryBase creation?

**Review by jmederosalvarado — COMMENTED** [review]
Overall good. I left a question. Let's make sure we run a node and send it a tx before merging.

**Comment by LukaszRozmej** [comment]
> Overall good. I left a question. Let's make sure we run a node and send it a tx before merging.

I think the tx will be ignored as the tx pool won't do anything with it

**Comment by benaadams** [comment]
> > Overall good. I left a question. Let's make sure we run a node and send it a tx before merging.
> 
> I think the tx will be ignored as the tx pool won't do anything with it

Should forward it to the sequencerUrl

**Comment by benaadams** [comment]
> > Overall good. I left a question. Let's make sure we run a node and send it a tx before merging.
> 
> I think the tx will be ignored as the tx pool won't do anything with it

Should forward it to the sequencerUrl; which is why no point in having the TxPool doing work on every new head

**Review by emlautarom1 — APPROVED** [review]

**Comment by emlautarom1** [comment]
Tested in OP-Sepolia through `eth_sendRawTransaction` and I'm getting responses from the sequencer as expected.

**Pull request merged**